sql/mozfun/google_ads/extract_country_code_from_campaign_name/udf.sql

```diff
@@ -0,0 +1,38 @@
+CREATE OR REPLACE FUNCTION google_ads.extract_country_code_from_campaign_name(
+  campaign_name STRING
+) AS (
+  CASE
+    WHEN REGEXP_CONTAINS(campaign_name, r"^(Mozilla_FF_UAC)")
+      THEN (REGEXP_EXTRACT(campaign_name, r"^Mozilla_FF_UAC_(?:[A-Z]{2}|MGFQ3)_([A-Z]{2})"))
+    WHEN REGEXP_CONTAINS(campaign_name, r"^(Mozilla_Firefox_)")
+      THEN (REGEXP_EXTRACT(campaign_name, r"^Mozilla_Firefox_([A-Z]{2})"))
+    WHEN REGEXP_CONTAINS(campaign_name, r"^(Mozilla_[A-Z]{2}_FF)")
+      THEN (
+      -- These campaigns targeted multiple countries/languages, so we don't know
+          NULL
+        )
+    ELSE NULL
+  END
+);
+
+SELECT
+  assert.equals(
+    google_ads.extract_country_code_from_campaign_name("Mozilla_FF_UAC_EU_AT_EN_AllGroups_Event1"),
+    "AT"
+  ),
+  assert.equals(
+    google_ads.extract_country_code_from_campaign_name("Mozilla_FF_UAC_MGFQ3_KE_EN_CPI"),
+    "KE"
+  ),
+  assert.equals(
+    google_ads.extract_country_code_from_campaign_name("Mozilla_FF_UAC_AU_AU_EN_AllGroups_Event7"),
+    "AU"
+  ),
+  assert.equals(
+    google_ads.extract_country_code_from_campaign_name("Mozilla_Firefox_DE_DE_Google_UAC_Android"),
+    "DE"
+  ),
+  assert.equals(
+    google_ads.extract_country_code_from_campaign_name("Mozilla_NA_FF_UAC_Group1_Event1"),
+    CAST(NULL AS STRING)
+  ),
```

Comment:
Should we add a test case for when null is passed? Are we sure campaign_name will either be a text value or null or could it be an empty string?

Reply:
Good point, I can add both.

I'm actually thinking I should consolidate these 3 UDFs into a single UDF that returns a struct.